Modules/_io/fileio.c

```diff
@@ -1229,16 +1249,27 @@ get_mode(fileio *self, void *closure)
     return PyUnicode_FromString(mode_string(self));
 }
 
+static PyObject *
+get_blksize(fileio *self, void *closure)
+{
+#ifdef HAVE_STRUCT_STAT_ST_BLKSIZE
+    if (self->stat_atopen != NULL && self->stat_atopen->st_blksize > 1) {
```

Comment:
I do wonder how realistic the st_blksize values, when available, are for performance purposes, I guess we'll find out.

Reply:
This PR should not change the buffer size, does it?

Reply:
https://github.com/python/cpython/issues/117151#issuecomment-2086343711 investigated `st_blksize` a bit previously. This PR I tried not to change buffer size at all / just change how it is accessed.

Have with the refactors + optimizations been watching for new issues. Are finding some as people test `main` (ex. gh-113977 which I wrote a primary fix for https://github.com/python/cpython/pull/122101, and have more fix ideas on top of the stat_atopen changes)